### PR TITLE
Updated Azure DevOps, DevTools, and Linguist to use `fetchApi` as per ADR013

### PR DESCRIPTION
**Pull request by awanlin**
## Hey, I just made a Pull Request!

Updated Azure DevOps, DevTools, and Linguist to use `fetchApi` as per [ADR013](https://backstage.io/docs/architecture-decisions/adrs-adr013)

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: plugins/azure-devops/src/api/AzureDevOpsClient.ts [hunk 1]
```diff
@@ -27,21 +27,28 @@ import {
...
+  FetchApi,
+  IdentityApi,
+} from '@backstage/core-plugin-api';
 import { ResponseError } from '@backstage/errors';
 import { AzureDevOpsApi } from './AzureDevOpsApi';
 
 /** @public */
 export class AzureDevOpsClient implements AzureDevOpsApi {
   private readonly discoveryApi: DiscoveryApi;
   private readonly identityApi: IdentityApi;
```
Comment (aramissennyeydd) [inline plugins/azure-devops/src/api/AzureDevOpsClient.ts:41]: If this is just a frontend plugin and we're not using this client in the backend, we shouldn't need the `identityApi`, that will get added automatically as part of the `fetchApi` middleware. https://github.com/backstage/backstage/blob/patch/v1.22.0/packages/core-app-api/src/apis/implementations/FetchApi/IdentityAuthInjectorFetchMiddleware.ts#L55-L77

Comment (awanlin) [reply plugins/azure-devops/src/api/AzureDevOpsClient.ts]: Yup, already done ;)

File: plugins/azure-devops/src/plugin.ts [hunk 2]
```diff
@@ -56,9 +57,13 @@ export const azureDevOpsPlugin = createPlugin({
   apis: [
     createApiFactory({
       api: azureDevOpsApiRef,
-      deps: { discoveryApi: discoveryApiRef, identityApi: identityApiRef },
-      factory: ({ discoveryApi, identityApi }) =>
-        new AzureDevOpsClient({ discoveryApi, identityApi }),
+      deps: {
+        discoveryApi: discoveryApiRef,
+        identityApi: identityApiRef,
```
[the remark this answers is not in the document]
Comment (awanlin) [reply plugins/azure-devops/src/plugin.ts]: Oh, good catch! Fixed 👍

**Review by freben — APPROVED** [review]
yes!

**Comment by awanlin** [comment]
Many thanks @aramissennyeydd and @freben 🚀

**Pull request merged**